src/devices/bus/nes/kaiser.cpp

```diff
@@ -778,15 +781,15 @@ void nes_ks7021a_device::write_h(offs_t offset, u8 data)
 u8 nes_ks7010_device::read_m(offs_t offset)
 {
 //  LOG_MMC(("ks7010 read_m, offset: %04x, data: %02x\n", offset, data));
-	return m_prg[m_latch * 0x2000 + offset];
+	return m_prg[(m_latch * 0x2000 + offset) & (m_prg_size - 1)];
```

Comment:
Is there anywhere you ensure (or even just assert) that `m_prg_size` is a power of two before doing this?

Reply:
No. Assuming a power of two is baked in to a lot of cases like this (2 other times in this file even). This one and many others are devices for unique bootlegs where the size is known and expected. The ANDing is just to avoid segfaulting on the off chance somebody throws a weird file at MAME that happens to have this one-off mapper number.

Reply:
It’s still a good idea to at least add an assert if it’s conceivably possible to get unexpected size media into it.